### PR TITLE
DAOS-7208 control: Update all vendored dependencies

### PR DESCRIPTION
**Pull request by mjmac**
- switch raft-boltdb to upstream v2
  - update raft to v1.2.0
  - update grpc to v1.37.0
  - update protobuf to v1.26.0
  - update yaml to v2.3.0

### CODE REVIEW

**Comment by mjmac** [comment]
Checkpatch failure is in vendored code. I thought that we had disabled scanning that stuff but I guess not...

File: src/control/security/grpc_authorization_test.go [hunk 1]
```diff
@@ -142,6 +142,9 @@ func TestSecurity_AllRpcsAreAuthorized(t *testing.T) {
 
 			var rpcNames []string
 			for i := 0; i < svcType.NumMethod(); i++ {
+				if strings.HasPrefix(svcType.Method(i).Name, "mustEmbedUnimplemented") {
```
Comment (kjacque) [inline src/control/security/grpc_authorization_test.go:145]: Curious to know where this comes from. I noticed the control and management services are using some protobuf unimplemented type now, and I'm guessing it has something to do with that. Would be nice to comment this with some background.

**Review by kjacque — COMMENTED** [review]
Nice work sorting all this out.

File: src/control/security/grpc_authorization_test.go (hunk 1 above)
Comment (mjmac) [reply src/control/security/grpc_authorization_test.go:145]: It's now a requirement that protobuf service implementations embed this struct in order to allow new methods to be added without requiring existing implementations to implement them. I'm having trouble finding a reference to it in the documentation though, so I guess most people encounter/resolve this when they try to build. https://github.com/grpc/grpc-go/issues/3669 has some discussion and links to the original issue/motivation. tl;dr: Adding this requirement brings the Go implementation into parity with implementations in other languages.

Comment (kjacque) [reply src/control/security/grpc_authorization_test.go:145]: Okay. So I guess that means this test ignores unimplemented methods, which makes sense to me.

**Review by kjacque — DISMISSED** [review]

File: src/control/drpc/modules_test.go [hunk 2]
```diff
@@ -9,22 +9,12 @@ package drpc
...
 	"testing"
 
-	"github.com/golang/protobuf/proto"
 	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/proto"
 
 	"github.com/daos-stack/daos/src/control/common"
 )
 
-func TestMarshal_Failed(t *testing.T) {
```
Comment (tanabarr) [inline src/control/drpc/modules_test.go:18]: why was this removed?

**Review by tanabarr — APPROVED** [review]

File: src/control/drpc/modules_test.go (hunk 2 above)
Comment (mjmac) [reply src/control/drpc/modules_test.go:18]: Doesn't fail anymore. Calling `proto.Marshal(nil)` just returns nil with no error.

**Review by dpquigl — APPROVED** [review]

**Comment by mjmac** [comment]
Sorry, had to force-push as merge wasn't playing well with the large change. No updates aside from resolving conflicts with latest master. Will try to get this landed early next week so that it doesn't drag on and run into more conflicts.

**Review by kjacque — DISMISSED** [review]

**Review by dpquigl — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]
